examples/parlai_chat_task_demo/webapp/src/app.jsx

```diff
@@ -164,34 +150,37 @@ function MainApp() {
   }
 
   return (
-    <div>
-      <BaseFrontend
-        task_config={taskConfig}
-        chat_state={chatState}
-        agent_id={agentId}
-        agent_status={agentStatus}
-        agent_state={agentState || {}}
-        connection_status={connectionStatus}
-        task_data={taskContext} // TODO fix naming issues - taskData is the initial data for a task, task_context may change through a task
-        messages={messages}
-        onMessageSend={(message) => {
-          message.id = agentId;
-          message.episode_done = agentState?.task_done || false; 
-          return sendMessage(message)
-            .then((msg) => {
-              addMessage(msg);
-            })
-            .then(() => setChatState("waiting"));
+    <MephistoContext.Provider value={mephistoProps}>
+      <AppContext.Provider
+        value={{
+          taskContext,
+          appSettings,
+          setAppSettings,
+          onTaskComplete: () => {
+            destroy();
+            handleSubmit({});
+          },
         }}
-        allDoneCallback={() => {
-          destroy();
-          handleSubmit({});
+      >
+        <div className="container-fluid" id="ui-container">
+          <BaseFrontend
+            inputMode={inputMode}
+            messages={messages}
+            onMessageSend={handleMessageSend}
```

Comment:
One question - why are these props rather than app context? I'm not sure when you should be writing one verses the other.

Reply:
Good question. This is largely subjective, and you can use either approach to send data down.

I chose these as the props in this case because I like the reduction of the app as a mental model as a simple chat container... it shows messages, it calls a callback on new message sends, and it has a prop to toggle it's "mode". All of the other props seemed "contextual" and not relevant to the core functionality of the app at a reductionist level, so I moved those to the app context. I can also see `onTaskComplete` being here but I kept it in the context since I found myself having to do a lot of prop drilling again to get it to the components that need it.